chain/mempool.go

```diff
@@ -11,74 +11,168 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+// txIndex defines the transaction index type. We use uint16 here as the max
+// allowed output index cannot exceed 65535.
+type txIndex uint16
```

Comment:
I think a transaction can have more than 65535 outputs if:
- it's non-standard
- it has many `OP_RETURN` outputs with small data pushes
- it takes up the entire block

For this reason, I don't think we can have this be a `uint16` because this type of transaction would cause a panic on all LND nodes. Another interesting tidbit is that this tx would be disallowed from being a channel open by LN if it included the funding output since it exceeds the 2-byte output index.

Reply:
Cool will move this into a different PR.

Reply:
Ah true about non-standard tricks (eg: you can only have a single `OP_RETURN` in a transaction with standardness). So we'll either need to bump up that integer size, or try to roll something a bit more custom (basically a series of byte arrays with custom limbs).